src/vs/workbench/contrib/terminal/browser/terminalMenus.ts

```diff
@@ -0,0 +1,332 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { localize } from 'vs/nls';
+import { MenuRegistry, MenuId } from 'vs/platform/actions/common/actions';
+import { ContextKeyAndExpr, ContextKeyEqualsExpr, ContextKeyExpr } from 'vs/platform/contextkey/common/contextkey';
+import { KEYBINDING_CONTEXT_TERMINAL_PROCESS_SUPPORTED, TERMINAL_COMMAND_ID, TERMINAL_VIEW_ID } from 'vs/workbench/contrib/terminal/common/terminal';
+
+const enum ContextMenuGroup {
+	Create = '1_create',
+	Edit = '2_edit',
+	Clear = '3_clear',
+	Kill = '4_kill',
+	Config = '5_config'
+}
+
+export const enum ContextMenuTabsGroup {
+	Default = '1_create_default',
+	Profile = '2_create_profile',
+	Configure = '3_configure'
+}
+
+export const enum ContextMenuTerminalGroup {
+	Create = '1_create',
+	Run = '2_run',
+	Manage = '3_manage',
+	Configure = '4_configure'
+}
+
+export function setupTerminalMenus(): void {
+	MenuRegistry.appendMenuItems(
```

Comment:
I've found the names of the menus a little confusing as we've added more. I was going to say let's add a header comment for each of these but how about we do a pass over the names instead?

Suggestions:

- MenubarTerminalMenu -> 👌 
- TerminalContainerContext -> TerminalInstanceContext
- TerminalToolbarContext -> TerminalNewDropdownContext
- TerminalTabsWidgetContext -> TerminalTabContext
- TerminalTabsWidgetEmptyContext -> TerminalTabEmptyAreaContext
- TerminalSingleTabContext -> TerminalInlineTabContext (can this be merged with TerminalTabContext?)
- TerminalTabInlineActions -> TerminalTabInlineActionContext

Reply:
done except there is no `TerminalTabInlineActions `anymore since you've moved that inside of the tabs widget